classes/security/Validation.php

```diff
@@ -474,6 +482,83 @@ public static function canAdminister($administeredUserId, $administratorUserId)
         // There were no conflicting roles. Permit administration.
         return true;
     }
+
+    /**
+     * Get the user's administration level
+     *
+     * @param int   $administeredUserId     User ID of user to potentially administer
+     * @param int   $administratorUserId    User ID of user who wants to do the administrating
+     * @param int   $contextId              The journal/context Id
+     * 
+     * @return int The authorized adminstration level 
+     */
+    public static function getAdministrationLevel(int $administeredUserId, int $administratorUserId, int $contextId = null): int
+    {
+        // You can administer yourself
+        if ($administeredUserId == $administratorUserId) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        $filteredSiteAdminUserGroups = Repo::userGroup()
+            ->getCollector()
+            ->filterByContextIds([\PKP\core\PKPApplication::CONTEXT_SITE])
+            ->filterByRoleIds([Role::ROLE_ID_SITE_ADMIN]);
+
+        // You cannot adminster administrators
+        if ($filteredSiteAdminUserGroups->filterByUserIds([$administeredUserId])->getCount() > 0) {
+            return self::ADMINISTRATION_PROHIBITED;
+        }
+
+        // Otherwise, administrators can administer everyone
+        if ($filteredSiteAdminUserGroups->filterByUserIds([$administratorUserId])->getCount() > 0) {
+            return self::ADMINISTRATION_FULL;
+        }
+
+        // Check for administered user group assignments in other contexts
+        // that the administrator user doesn't have a manager role in.
+        $userGroupsCount = Repo::userGroup()
+            ->userUserGroups($administeredUserId)
+            ->filter(fn($userGroup) => 
+                $userGroup->getContextId() != \PKP\core\PKPApplication::CONTEXT_SITE &&
+                !Repo::userGroup()
+                    ->getCollector()
+                    ->filterByContextIds([$userGroup->getContextId()])
+                    ->filterByUserIds([$administratorUserId])
+                    ->filterByRoleIds([Role::ROLE_ID_MANAGER])
+                    ->getCount()
+            )
+            ->count();
+        
+        if ( $userGroupsCount > 0 ) {
+            // Found an assignment: disqualified.
+            // But also determine if a partial administrate is allowed
+            // if the Administrator User is a Journal Manager in the current context
+            if ($contextId && 
```

Comment:
I would suggest checking `$contextId !== null`, to distinguish from e.g. `CONTEXT_SITE` (which would be `0`).

Reply:
done.